### ISSUE TITLE
Thinking real64 is a package

### ISSUE THREAD

**Issue by willelm** [LLNL/spack#26288]
### Steps to reproduce

```console
$ spack install seacas
```

### Error message

```console
$ ==> Error: variant "real64" not found in package "parmetis" [required from package "seacas"]
```

### Information on your system

This is on LLNL rztrona. I think it's just that the concretizer thinks that real64 is it's own package when it's really just an option to metis. 

* **Spack:** 0.16.3-4487-87450f3688
* **Python:** 3.7.2
* **Platform:** linux-rhel7-broadwell
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by haampie** [comment]
The error message really tells you exactly what's wrong in seacas...

**Comment by willelm** [comment]
OK, so this is on the maintainer of seacas. @gsjaardema . We were able to build seacas by deleting any reference to +int64 and +real64 in the package setup file.

**Comment by gsjaardema** [comment]
I have (actually before seeing this...) been working on a new tag of seacas and noticed the metis/parmetis issue. I should hopefully have a PR up soon that will fix this.

**Comment by gsjaardema** [comment]
should be fixed by #26310
